### PR TITLE
Add search field to creative inventory

### PR DESCRIPTION
**Pull request by kilbith**
You'll no longer waste your time browsing several pages to find something.
Obviously gives a significant help to the creative activity.

screenshot_20151129_222037

screenshot_20151129_221146

### CODE REVIEW

**Comment by BlockMen** [comment]
```2015-12-01 17:37:49: ERROR[Main]: ModError: Failed to load and run script from ..\games\minetest_next\mods\creative\init.lua:
2015-12-01 17:37:49: ERROR[Main]: ..\games\minetest_next\mods\creative\init.lua:57: ')' expected (to close '(' at line 7) near 'creative_inv_update'```

**Comment by kilbith** [comment]
Oopsie, small mistake. Fixed and squashed.

**Comment by BlockMen** [comment]
Search filter is applied to all connected players.

**Comment by kilbith** [comment]
Would you agree if a detached inventory is created per player ? That's the only way I think.

**Comment by BlockMen** [comment]
Seems so, yes. Do it ;)

**Comment by LNJplus** [comment]
Nice!
And could you add a place for armor, please! :smiley:

**Comment by kilbith** [comment]
Armor makes sense only in survival mode.

**Comment by LNJplus** [comment]
But maybe you want to take armor off or you want to look different.
Okay for this we could add a command to open an armor menu. :smirk_cat:

**Comment by kilbith** [comment]
@BlockMen Rebased and squashed. Inventory's filtering is player-independant now.

File: mods/creative/init.lua [hunk 1]
```diff
@@ -35,21 +40,29 @@ minetest.after(0, function()
 			end
 		end,
 	})
+
+	creative_inventory.creative_inv_update(player_name, nil)
+	--print("creative inventory size: "..dump(creative_inventory.creative_inventory_size))
+end
+
+function creative_inventory.creative_inv_update(player_name, filter)
```
Comment (BlockMen) [inline mods/creative/init.lua]: Why not ```creative_inventory.update(player_name, filter)``` ?

File: mods/creative/init.lua [hunk 2]
```diff
@@ -35,21 +40,29 @@ minetest.after(0, function()
...
 	local creative_list = {}
-	for name,def in pairs(minetest.registered_items) do
-		if (not def.groups.not_in_creative_inventory or def.groups.not_in_creative_inventory == 0)
-				and def.description and def.description ~= "" then
-			table.insert(creative_list, name)
+	local inv = minetest.get_inventory({type="detached", name="creative_"..player_name})
```
Comment (BlockMen) [inline mods/creative/init.lua]: Please add blanks around ```..```, ```=``` and the like.

**Comment by BlockMen** [comment]
Except the style issues it seems fine now :+1:

**Comment by kilbith** [comment]
Style fixed.

**Comment by kilbith** [comment]
Btw I'm not sure we need `tostring(pagenum)` and `tostring(pagemax)`, since both are inherently numbers...

**Comment by BlockMen** [comment]
> Armor makes sense only in survival mode.

Same goes for crafting. So either both or none.

**Comment by kilbith** [comment]
Yes so the craft grid should be trashed out in creative mode.

**Pull request merged**